tycho-core/src/main/java/org/eclipse/tycho/core/maven/MavenDependencyInjector.java

```diff
@@ -126,17 +134,34 @@ private boolean requiresExternalDependencies() {
         return bundleReader.loadManifest(bundleLocation).getBundleClasspath();
     }
 
-    private Dependency createSystemScopeDependency(ArtifactKey artifactKey, File location) {
-        return createSystemScopeDependency(artifactKey, P2_GROUPID_PREFIX + artifactKey.getType(), location);
+    private Dependency createSystemScopeDependency(ArtifactDescriptor descriptor, File location) {
+        ArtifactKey artifactKey = descriptor.getKey();
+        return createSystemScopeDependency(descriptor, P2_GROUPID_PREFIX + artifactKey.getType(), location);
     }
 
-    private Dependency createSystemScopeDependency(ArtifactKey artifactKey, String groupId, File location) {
+    private Dependency createSystemScopeDependency(ArtifactDescriptor descriptor, String groupId, File location) {
         Dependency dependency = new Dependency();
-        dependency.setGroupId(groupId);
-        dependency.setArtifactId(artifactKey.getId());
-        dependency.setVersion(artifactKey.getVersion());
-        dependency.setScope(Artifact.SCOPE_SYSTEM);
-        dependency.setSystemPath(location.getAbsolutePath());
+        MavenDependencyDescriptor dependencyDescriptor = descriptorMapping == null ? null
+                : descriptorMapping.apply(descriptor);
+        if (dependencyDescriptor != null) {
```

Comment:
Is dependency descriptor set on many artifacts, including some that do not come from Maven? Shouldn't we also check whether location belongs to Maven repo?